apps/prairielearn/src/middlewares/autoEnroll.ts

```diff
@@ -35,8 +41,25 @@ export default asyncHandler(async (req, res, next) => {
 
   // Check if the self-enrollment institution restriction is satisfied
   const institutionRestrictionSatisfied =
-    !courseInstance.self_enrollment_restrict_to_institution ||
-    res.locals.authn_user.institution_id === res.locals.course.institution_id;
+    res.locals.authn_user.institution_id === res.locals.course.institution_id ||
+    // If enrollment management is not yet enabled, instructors have no way to
+    // configure this restriction, so we'll ignore it. This is critical for workflows
+    // where courses use `institution: Any` in course instance access rules to permit
+    // non-institution users to access course instances.
+    //
+    // Note that skipping this check won't arbitrarily allow non-institution users to
+    // access course instances. We still rely on course instance access rules to gate
+    // access for users outside of a specific institution. If those access rules aren't
+    // satisfied, the user won't get as far as this middleware.
+    //
+    // TODO: we need to reconsider this before enrollment management is enabled by default.
+    // Specifically, we need to consider what'll happen when a course instance has
+    // enrollment management enabled and is still using legacy access rules, and specifically
+    // those with `institution: Any`. In that case, there would be effectively two ways to
+    // control institution self-enrollment restrictions: via access rules, and via the
+    // self-enrollment restriction flag. This could be confusing.
```

Comment:
@reteps I'd like you to think about this and propose a solution.

Reply:
We had discussed an explicit flag to switch between old and new systems of access control. Maybe we should also do this for enrollment management as well? I feel like this would be clearer in code (we can branch on this flag) and also for the user (we can show them a "convert to new system" button, with a warning that the old system will be going away on date XXX).